### ISSUE TITLE
short circuit conditionals are identified as suspect

### ISSUE THREAD

**Issue by jlapeyre** [tonyhffong/Lint.jl#73]
Something like

```julia
if z == 0
   k == 0 && return z
   ...
```
Gives
> The 1st statement under the true-branch is a boolean expression. Typo?

And (in this case)
> Comparing apparently incompatible types (#1) LHS:Complex{T<:Real} RHS:Int64

--John

**Comment by tonyhffong** [comment]
Yes, this code is too cute for Lint in its current capabilities.

In the first case, it tries to check with you if you actually have omitted a `||` or `&&` in the conditional line. 

Lint expects both left-hand-side and right-hand-side of && to be bool, and would be "helpful" if it can prove otherwise.

**Comment by tonyhffong** [comment]
Try the current master to see if the new behavior is more reasonable.

**Comment by jlapeyre** [comment]
Hi, thanks for the note.  It's been so long since I looked, I thought this was about a bug in Coverage; retested that. Then I reread your message. I hope I pulled the correct branch of Lint, it is master.
`f1c06df51d99c7da61b25039a50c8c7a95bb6e8a`
Wed Feb 11 20:08:10 2015 +0700

I am still getting what  seem to be spurious messages. The file I am linting is
https://github.com/jlapeyre/LambertW.jl/blob/master/src/lambertw.jl
And this is the output, which gives warnings for short circuit `&&` and `||`. (as well as some correct warnings.)

```
home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:51 [lambertwk0     ] ERROR  one: ErrorException("no unique matching method for the specified argument types")
                                                            Signature: (DataType,)

/home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:68 [_lambertwkm1   ] ERROR  one: ErrorException("no unique matching method for the specified argument types")
                                                            Signature: (DataType,)

/home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:69 [_lambertwkm1   ] ERROR  one: ErrorException("no unique matching method for the specified argument types")
                                                            Signature: (DataType,)

/home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:86 [lambertw       ] ERROR  Use of undeclared symbol LambertW

/home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:35 [_lambertw      ] WARN   Comparing apparently incompatible types (#1) LHS:Float64 RHS:Function

/home/jlapeyre/.julia/v0.4/LambertW/src/lambertw.jl:98 [lambertw       ] WARN   Comparing apparently incompatible types (#1) LHS:Complex{T<:Real} RHS:Int64``
```

**Comment by tonyhffong** [comment]
The one on `lambertw.jl:35` has a line count issue. It should be referring to the condition `lastdiff == diff` where `diff` is a standard julia function.

**Comment by jlapeyre** [comment]
Ah, I see. If you need any more testing, let me know.
